src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -143,28 +143,21 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
             {
                 try
                 {
-                    (IPackageReferenceContextInfo TopLevelPackage, IPackageReferenceContextInfo TransitivePackage) installedPackage = await GetInstalledAndTransitivePackagesAsync(project.NuGetProject, Id, cancellationToken);
-                    var topLevelPackageVersion = installedPackage.TopLevelPackage;
-                    var transitivePackageVersion = installedPackage.TransitivePackage;
+                    IPackageReferenceContextInfo packageContext = await GetPackageContextAsync(project.NuGetProject, Id, cancellationToken);
 
                     project.InstalledVersionMaxVulnerability = -1;
                     project.RequestedVersion = null;
                     project.AutoReferenced = false;
 
-                    if (transitivePackageVersion != null)
+                    if (packageContext == null)
                     {
-                        project.PackageLevel = PackageLevel.Transitive;
-                        UpdateProjectInstallationInfo(project, transitivePackageVersion, installedVersionsSet);
-                    }
-                    else if (topLevelPackageVersion != null)
-                    {
-                        project.PackageLevel = PackageLevel.TopLevel;
-                        UpdateProjectInstallationInfo(project, topLevelPackageVersion, installedVersionsSet);
+                        project.InstalledVersion = null;
+                        project.PackageLevel = null;
                     }
                     else
                     {
-                        project.InstalledVersion = null;
-                        project.PackageLevel = null;
+                        project.PackageLevel = packageContext is ITransitivePackageReferenceContextInfo ? PackageLevel.Transitive : PackageLevel.TopLevel;
```

Comment:
May be out of scope for this PR. Looking at the `is` operator, I think `ITransitivePackageReferenceContextInfo` interface contract can be improved. Here are the reasons.
- AFAIK, there is a concept called "Transitive package" but no such thing called "Transitive package reference" in NuGet.
- `ITransitivePackageReferenceContextInfo` inherting `IPackageReferenceContextInfo` interface seems incorrect to me because it is primarily meant for obtaining "Transitive origins" for a package.

```cs
public interface IPackageReferenceContextInfo
{
    PackageIdentity Identity { get; }
    NuGetFramework? Framework { get; }
    VersionRange? AllowedVersions { get; }
    bool IsAutoReferenced { get; }
    bool IsUserInstalled { get; }
    bool IsDevelopmentDependency { get; }
    VersionRange? VersionOverride { get; }
}

public interface ITransitivePackageContextInfo
{
    PackageIdentity Identity { get; }
    IEnumerable<IPackageReferenceContextInfo> TransitiveOrigins { get; }
}
``` 
I am open for feedback and happy to create a tech debt tracking issue.

Reply:
Definitely out of scope for this refactor. Feel free to open an issue and explain what alternative you propose. Thanks!

Reply:
Created tracking tech debt issue https://github.com/NuGet/Client.Engineering/issues/3032